Feature/table implementation

mpgxvii opened the pull request.

herkulano commented on src/app/components/tables/patient-table/patient-table.component.ts:
```diff
@@ -0,0 +1,54 @@
+
+import { Subject } from '../../../shared/store/subject/subject.model'
+
+export class PatientDataSource extends DataSource<any> {
```
all `patient` names should be `subject` instead, for consistency.

herkulano commented on src/app/components/tables/patient-table/patient-table.component.ts:
```diff
@@ -0,0 +1,54 @@
+export class PatientDataSource extends DataSource<any> {
+  mydata: Subject[]
+
+  constructor (private subjects: Observable<Subject[]>, private _paginator: MdPaginator) {
```
remove `_` from private variables, for consistency, unless there are getters and setters.

herkulano commented on src/app/components/tables/patient-table/patient-table.component.ts:
```diff
@@ -0,0 +1,54 @@
+import { Subject } from '../../../shared/store/subject/subject.model'
+
+export class PatientDataSource extends DataSource<any> {
+  mydata: Subject[]
```
change `mydata` to `data`. `mysomething` is used in tutorials and examples, it should not be used in production code.

herkulano commented on src/app/components/tables/patient-table/patient-table.component.ts:
```diff
@@ -0,0 +1,54 @@
+  disconnect () {}
+}
+
+@Component({
```
Move component to the top for clarity, to understand/read the component first.

mpgxvii replied:
Hi, I tried doing this but it gives me a tslint error saying "PatientDataSource is used before declaration" since I'm using it in PatientTableComponent? Or is that fine?

herkulano replied:
Good point, let's move it to another file then.

herkulano commented on src/app/components/tables/patient-table/patient-table.component.ts:
```diff
@@ -0,0 +1,54 @@
+
+  constructor (private subjects: Observable<Subject[]>, private _paginator: MdPaginator) {
+    super()
+    this.subjects.subscribe(res => this.mydata = res)
```
`this.subjects.subscribe` is unnecessary. instead, you can use the  `| async` pipe in the template to subscribe to the Observable, it makes the code a lot cleaner and the `| async` pipe takes also care of unsubscribing when the component is destroyed.

https://angular.io/api/common/AsyncPipe

herkulano replied:
`res` is short for response, which is used when there's a data response from an http call, use `d` or `x` for generic data.

herkulano commented on src/app/pages/study/study.component.html:
```diff
@@ -0,0 +1,68 @@
+    <div class="content">
+    <p *ngIf="!(isLoaded$ | async)">Loading...</p>
+      <ng-container *ngIf="isStudyLoadedAndValid$ && isLoaded$ | async">
+        <!--<p *ngFor="let subject of (subjects$ | async)">
```
avoid leaving commented code behind, remove it completely. if we need it back for some reason we use the git's history.

mpgxvii commented:
Hi, I've added the store for getting data for the device status and the tooltip for it in the patient cohort table. 

I have a few issues and questions: 
1. There are duplicate sources for subjects so it is difficult to get the entities in the store to have the id:entity format. 
2. Currently, it's just getting the status data. Should I get more data such as the data for the data types and compliance chart (in the table)? Should I create an interface for the data taken from the API for the table?
3. Currently, I am dispatching an action for each source of each subject (in the device status component). Will I have to dispatch a different action for other data (in a separate component) or merge them into one dispatch the action from the parent subject table component for each patient instead?
4. I've added some tests but they're failing. It says I should consider moving the components to a higher module, should I create modules for the device tooltip components I created?

Let me know what I should change or if I did things incorrectly. 
Thank you!
